### PR TITLE
Html pdf framework

### PR DESCRIPTION
**Pull request by manojkrsaini09**
added services to 1. generate html 
                            2.convert html to pdf
added same sample demo service to show how we will use conversion and template services.

### CODE REVIEW

File: src/main/java/creativei/controller/MainController.java [hunk 1]
```diff
@@ -107,10 +118,12 @@ public Boolean uploadLocalityDate() throws IOException {
 	
 
 	// Mail ids for demo mails
-	public static final String mailID1 = "";
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: Use properties file for this. Move this to properties file and fetch id from there

File: src/main/java/creativei/controller/MainController.java [hunk 2]
```diff
@@ -170,4 +184,59 @@ public void getTestHtml() {
 		    String html = templateService.generateHTML("VMDemo.vm", "templates/", ctx);
 		    logger.debug(html);
 	}
+	
+	
+	// service to show generate  and download pdf
+	@RequestMapping(value = "/generateDownloadPdf" , method = RequestMethod.GET)
+	public void downloadPdf(HttpServletResponse res) {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: Is this method just for testing?

File: src/main/java/creativei/controller/MainController.java [hunk 3]
```diff
@@ -170,4 +184,59 @@ public void getTestHtml() {
...
+		public void generateSendPdf(HttpServletResponse res) {
+			    Map<String,Object> ctx = new HashMap<String,Object>();
+			    ctx.put("name", "Manoj");
+			    ctx.put("lastName", "Saini");
+			    String html = templateService.generateHTML("VMDemo.vm", "templates/", ctx);
+			    ByteArrayOutputStream  outputStream = new ByteArrayOutputStream();
+			    pdfConversionService.convertHtmlToPdf(outputStream, html);
+			    byte[] bytes = outputStream.toByteArray();
+			
+			    MailObject email = new MailObject();
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: We should have a method in mailService that would do all this job behind the hood

File: src/main/java/creativei/service/PdfService/ItextConversionService.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+	@Override
+	public void convertHtmlToPdf(OutputStream outputStream, String html) {
+
+		// Create new document
+		Document document = new Document();
+		PdfWriter pdfWriter;
+
+		try {
+			pdfWriter = PdfWriter.getInstance(document,outputStream);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/PdfService/ItextConversionService.java]: Please check if we need to synchronize pdfWriter

File: src/main/java/creativei/service/PdfService/ItextConversionService.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+			document.open();
+			
+			//write html content to pdf file and then output stream
+			XMLWorkerHelper.getInstance().parseXHtml(pdfWriter, document,new ByteArrayInputStream(html.getBytes(StandardCharsets.UTF_8))); 
+			document.close();
+		} catch (FileNotFoundException | DocumentException e) {
+			e.printStackTrace();
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/PdfService/ItextConversionService.java]: use logger

File: src/main/java/creativei/util/CommonUtils.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
+package creativei.util;
+
+import java.io.IOException;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CommonUtils {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/util/CommonUtils.java]: Should rename to PdfUtils or DocumentUtils as we are only keeping pdf/document related methods here.

File: src/main/java/creativei/controller/MainController.java (hunk 3 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/controller/MainController.java]: I think mail service should independent of conversion logic. We can send any type of file in mail. So create required file and just add file as attachment.

File: src/main/java/creativei/service/PdfService/ItextConversionService.java (hunk 5 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/service/PdfService/ItextConversionService.java]: done

File: src/main/java/creativei/util/CommonUtils.java (hunk 6 above)
Comment (manojkrsaini09) [reply src/main/java/creativei/util/CommonUtils.java]: In this file's current methods we are processing output stream to prepare download file. We can use this file to keep other common method also. So should we rename it?

File: src/main/java/creativei/controller/MainController.java (hunk 1 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/controller/MainController.java]: Ideally, even test mail ids should be picked from the properties file to speed up testing process and to follow standards. approving this one for now.

File: src/main/java/creativei/controller/MainController.java (hunk 3 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/controller/MainController.java]: Yes, but all work related to add mail body, recipient list etc should go to mail service

File: src/main/java/creativei/util/CommonUtils.java (hunk 6 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/util/CommonUtils.java]: Rename to DocumentUtils

**Pull request merged**